src/util/http_utils.js

```diff
@@ -664,6 +665,131 @@ function set_amz_headers(req, res) {
     res.setHeader('x-amz-id-2', req.request_id);
 }
 
+/**
+ * set_expiration_header sets the `x-amz-expiration` response header for GET, PUT, or HEAD object requests
+ * if the object matches any enabled bucket lifecycle rule
+ *
+ * @param {Object} req
+ * @param {http.ServerResponse} res
+ * @param {Object} object_md
+ */
+async function set_expiration_header(req, res, object_md) {
+    const rules = req.params.bucket && await req.object_sdk.get_bucket_lifecycle_configuration_rules({ name: req.params.bucket });
+    if (!object_md) { // calculating object_md for putObject
```

Comment:
from what I understand this only happens for put commands. its better that you construct this outside this function and pass the object_md as a parameter. also you should get as much information as you can from the response and not from the request

Reply:
Done, Thanks.